<!--

	Patricia Tavares
	patricia.tavares@cimat.mx
	Launch file to run multiple nodes containing the controllers

-->
<launch>
  <arg name="enable_logging" default="false"/>
  <arg name="node_name" default="formation_control" />

<!--
	Parameters area, modify your desired params here:
-->

  <arg name="n" default="3" />

  <arg name="control" default="7"/>
  <arg name="Kv" default="0.25"/>
  <arg name="Kw" default="0.75"/>

  <arg name="e_t" default="0.015"/>
  <arg name="e_psi" default="0.01"/>
  <arg name="max_time" default="15"/>

  <arg name="desired_formation" default="0" />
  <arg name="width" default="2" />

  <arg name="input_dir" default="/home/hector/patty_ws/src/formation_control/src/input/" />
  <arg name="output_dir" default="/home/hector/patty_ws/src/formation_control/src/output/" />
  <arg name="gamma_file" default="0"/>
  <arg name="matching" default="0"/>
  <arg name="communication_type" default="1"/>

<!--

	Do not modify this part, unless you want to add a new node run. For example, run 4 nodes instead of 3.

-->

  <group ns="$(arg node_name)0">
    <node name="$(arg node_name)" pkg="$(arg node_name)" type="$(arg node_name)" args="0 $(arg n) $(arg control) $(arg Kv) $(arg Kw) $(arg e_t) $(arg e_psi) $(arg max_time) $(arg desired_formation) $(arg width) $(arg input_dir) $(arg output_dir) $(arg gamma_file) $(arg matching) $(arg communication_type)" output="screen"/>
  </group>

  <group ns="$(arg node_name)1">
    <node name="$(arg node_name)" pkg="$(arg node_name)" type="$(arg node_name)" args="1 $(arg n) $(arg control) $(arg Kv) $(arg Kw) $(arg e_t) $(arg e_psi) $(arg max_time) $(arg desired_formation) $(arg width) $(arg input_dir) $(arg output_dir) $(arg gamma_file) $(arg matching) $(arg communication_type)" output="screen"/>
  </group>

  <group ns="$(arg node_name)2">
    <node name="$(arg node_name)" pkg="$(arg node_name)" type="$(arg node_name)" args="2 $(arg n) $(arg control) $(arg Kv) $(arg Kw) $(arg e_t) $(arg e_psi) $(arg max_time) $(arg desired_formation) $(arg width) $(arg input_dir) $(arg output_dir) $(arg gamma_file) $(arg matching) $(arg communication_type)" output="screen"/>
  </group>

</launch>
